src/games/strategy/engine/data/GameParser.java

```diff
@@ -835,21 +835,11 @@ private void parseAlliances(final Element root) throws GameParseException {
       // iterate through all players to get known allies and enemies
       for (final PlayerID currentPlayer : players) {
         // start with all players as enemies
-        final HashSet<PlayerID> enemies = new HashSet<>(players);
         // start with no players as allies
-        final HashSet<PlayerID> allies = new HashSet<>();
-        // iterate through all alliances the player is in
-        if (allianceTracker.getAlliancesMap().get(currentPlayer) != null) {
-          for (final String alliance : allianceTracker.getAlliancesMap().get(currentPlayer)) {
-            // iterate through the members of the alliances
-            for (final PlayerID alliedPlayer : allianceTracker.getPlayersInAlliance(alliance)) {
-              // add each allianceMember to the alliesList
-              allies.add(alliedPlayer);
-              // remove each allianceMember from the enemiesList
-              enemies.remove(alliedPlayer);
-            }
-          }
-        }
+        final Set<PlayerID> allies = allianceTracker.getAllies(currentPlayer);
+        final Set<PlayerID> enemies = new HashSet<>(players);
+        enemies.removeAll(allies);
```

Comment:
Not yet, at least at this juncture. The immediate goal was to remove the data implementation leak in alliance tracker, namely get it so the only thing using the alliance map was the AllianceTracker. Alllows for the conversion from map to multimap to be class internal.

Looking at the full block of code:
```
        final Set<PlayerID> allies = allianceTracker.getAllies(currentPlayer);
        final Set<PlayerID> enemies = new HashSet<>(players);
        enemies.removeAll(allies);

        // remove self from enemieslist (in case of free-for-all)
        enemies.remove(currentPlayer);
        // remove self from allieslist (in case you are a member of an alliance)
        allies.remove(currentPlayer);
        // At this point enemies and allies should be set for this player.
        for (final PlayerID alliedPLayer : allies) {
          relationshipTracker.setRelationship(currentPlayer, alliedPLayer,
              relationshipTypeList.getDefaultAlliedRelationship());
        }
        for (final PlayerID enemyPlayer : enemies) {
          relationshipTracker.setRelationship(currentPlayer, enemyPlayer,
              relationshipTypeList.getDefaultWarRelationship());
        }
```
It appears what should be happening is the alliance tracker probably should be a member of the relationship tracker. From there, it might make sense moving more stuff into the alliance tracker, but I do not want to make that decision at this moment just yet without being prepared to yet further increase scope.